src/Symfony/Component/Asset/Preload/HttpFoundationPreloadManager.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\Preload;
+
+use Symfony\Component\Asset\Exception\InvalidArgumentException;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Preload manager for the HttpFoundation component.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class HttpFoundationPreloadManager implements PreloadManagerInterface
+{
+    private $resources = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addResource($uri, $as = '', $nopush = false)
+    {
+        $this->resources[$uri] = array('as' => $as, 'nopush' => $nopush);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getResources()
+    {
+        return $this->resources;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setResources(array $resources)
+    {
+        foreach ($resources as $key => $options) {
+            if (!is_string($key)) {
+                throw new InvalidArgumentException('The key must be a path to an asset, "%s" given.', $key);
+            }
+
+            if (!isset($options['as']) || !is_string($options['as'])) {
+                throw new InvalidArgumentException('The "as" option is mandatory and must be a string.');
+            }
+
+            if (!isset($options['nopush']) || !is_bool($options['nopush'])) {
+                throw new InvalidArgumentException('The "nopush" option is mandatory and must be a bool.');
+            }
+        }
+
+        $this->resources = $resources;
+    }
+
+    /**
+     * Sets preload Link HTTP header.
+     *
+     * @param Response $response
+     */
+    public function setLinkHeader(Response $response)
```

Comment:
a more reusable API would be to have a method returning the header instead, without dependency on the Response. The event listener would then handle the binding to HttpFoundation by setting the header (allowing the same manager to be reused by people using a PSR-7 stack instead, without reimplementing all the logic in a different class just because of this method).
Currently, the most important part of the business logic is in this method, which is not part of the interface